plugin-interface/src/main/scala/mesosphere/marathon/plugin/optfactory/AppOptFactory.scala

```diff
@@ -0,0 +1,7 @@
+package mesosphere.marathon.plugin.optfactory
+
+import mesosphere.marathon.plugin.AppDefinition
+import mesosphere.marathon.plugin.plugin._
+
+/** AppOptFactory generates functional options that configure objects of type T */
+trait AppOptFactory[T] extends Opt.Factory[AppDefinition, T] with Plugin
```

Comment:
Is it worth declaring something so specific? or should we instead declare an `OptFactoryPlugin[P,T] extends Opt.Factory[P,T] with Plugin` ?